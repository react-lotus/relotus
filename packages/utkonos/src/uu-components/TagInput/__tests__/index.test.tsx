/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react';
import { TagInput } from '..';

const onChange = jest.fn();

const SIMPLE_VALUES = ['foo', 'bar', 'baz'];

describe('UIKIT | TagInput', () => {
  describe('Snapshots', () => {
    it('should default', () => {
      const tree = renderer
        .create(
          <TagInput id="tagInput" value={SIMPLE_VALUES} name="testName" onChange={onChange} />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should labeled row', () => {
      const tree = renderer
        .create(
          <TagInput
            id="tagInput"
            value={SIMPLE_VALUES}
            name="testName"
            label="labeled"
            onChange={onChange}
          />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should labeled column', () => {
      const tree = renderer
        .create(
          <TagInput
            id="tagInput"
            value={SIMPLE_VALUES}
            name="testName"
            label="labeled"
            direction="column"
            onChange={onChange}
          />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should with error', () => {
      const tree = renderer
        .create(
          <TagInput
            id="tagInput"
            value={SIMPLE_VALUES}
            name="testName"
            error="Test error"
            onChange={onChange}
          />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should disabled', () => {
      const tree = renderer
        .create(
          <TagInput
            id="tagInput"
            value={SIMPLE_VALUES}
            name="testName"
            disabled
            onChange={onChange}
          />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Reset button', () => {
    it('should expose disabled reset with disabled TagInput', () => {
      render(
        <TagInput
          id="tagInput"
          value={SIMPLE_VALUES}
          name="testName"
          disabled
          onChange={onChange}
        />,
      );
      const reset = screen.queryByText('Сбросить значения');
      expect(reset).toBeInTheDocument();
      if (reset) {
        expect(reset.closest('.uu-button')).toBeDisabled();
      }
    });
    it('should expose disabled reset with empty values', () => {
      render(<TagInput id="tagInput" value={[]} name="testName" onChange={onChange} />);
      const reset = screen.queryByText('Сбросить значения');
      if (reset) {
        expect(reset.closest('.uu-button')).toBeDisabled();
      }
    });
    it('should expose enabled reset', () => {
      render(
        <TagInput
          id="tagInput2"
          value={SIMPLE_VALUES}
          name="testName2"
          onChange={onChange}
          resetText="Очистить"
        />,
      );
      expect(screen.queryByText('Сбросить значения')).not.toBeInTheDocument();
      const reset2 = screen.queryByText('Очистить');
      expect(reset2).toBeInTheDocument();
      if (reset2) {
        expect(reset2.closest('.uu-button')).toBeEnabled();
        fireEvent.click(reset2);
      }
      expect(onChange).toHaveBeenCalledWith([]);
    });
  });
});
