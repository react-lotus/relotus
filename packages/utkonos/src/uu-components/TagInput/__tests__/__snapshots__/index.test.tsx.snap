// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UIKIT | TagInput Snapshots should default 1`] = `
<div>
  <div
    className="uu-tagInput"
    onClick={[Function]}
    role="none"
  >
    <span
      className="uu-multiSelectValue"
    >
      <div
        className="uu-multiSelectValueText"
      >
        <span
          className="uu-tagInputTagValue"
        >
          foo
        </span>
      </div>
      <button
        className="uu-multiSelectValueClose"
        onClick={[Function]}
        type="button"
      >
        <svg>
          icon-close.svg
        </svg>
      </button>
    </span>
    <span
      className="uu-multiSelectValue"
    >
      <div
        className="uu-multiSelectValueText"
      >
        <span
          className="uu-tagInputTagValue"
        >
          bar
        </span>
      </div>
      <button
        className="uu-multiSelectValueClose"
        onClick={[Function]}
        type="button"
      >
        <svg>
          icon-close.svg
        </svg>
      </button>
    </span>
    <span
      className="uu-multiSelectValue"
    >
      <div
        className="uu-multiSelectValueText"
      >
        <span
          className="uu-tagInputTagValue"
        >
          baz
        </span>
      </div>
      <button
        className="uu-multiSelectValueClose"
        onClick={[Function]}
        type="button"
      >
        <svg>
          icon-close.svg
        </svg>
      </button>
    </span>
    <input
      className="uu-tagInputNewItem"
      id="tagInput"
      name="testName"
      onBlur={[Function]}
      onKeyDown={[Function]}
      onPaste={[Function]}
    />
  </div>
  <button
    className="uu-button uu-button_view_dangerLink"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <svg>
      reset-icon.svg
    </svg>
    <span>
      Сбросить значения
    </span>
  </button>
</div>
`;

exports[`UIKIT | TagInput Snapshots should disabled 1`] = `
<div>
  <div
    className="uu-tagInput"
    onClick={[Function]}
    role="none"
  >
    <span
      className="uu-multiSelectValue uu-multiSelectValue_disabled"
    >
      <div
        className="uu-multiSelectValueText"
      >
        <span
          className="uu-tagInputTagValue"
        >
          foo
        </span>
      </div>
      <button
        className="uu-multiSelectValueClose uu-multiSelectValueClose_disabled"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        <svg>
          icon-close.svg
        </svg>
      </button>
    </span>
    <span
      className="uu-multiSelectValue uu-multiSelectValue_disabled"
    >
      <div
        className="uu-multiSelectValueText"
      >
        <span
          className="uu-tagInputTagValue"
        >
          bar
        </span>
      </div>
      <button
        className="uu-multiSelectValueClose uu-multiSelectValueClose_disabled"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        <svg>
          icon-close.svg
        </svg>
      </button>
    </span>
    <span
      className="uu-multiSelectValue uu-multiSelectValue_disabled"
    >
      <div
        className="uu-multiSelectValueText"
      >
        <span
          className="uu-tagInputTagValue"
        >
          baz
        </span>
      </div>
      <button
        className="uu-multiSelectValueClose uu-multiSelectValueClose_disabled"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        <svg>
          icon-close.svg
        </svg>
      </button>
    </span>
    <input
      className="uu-tagInputNewItem"
      disabled={true}
      id="tagInput"
      name="testName"
      onBlur={[Function]}
      onKeyDown={[Function]}
      onPaste={[Function]}
    />
  </div>
  <button
    className="uu-button uu-button_view_dangerLink"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    <svg>
      reset-icon.svg
    </svg>
    <span>
      Сбросить значения
    </span>
  </button>
</div>
`;

exports[`UIKIT | TagInput Snapshots should labeled column 1`] = `
<div
  className="uu-labeled_direction_column"
>
  <label
    aria-label="labeled"
    className="uu-labeledLabel_direction_column uu-label uu-label_text"
    htmlFor="tagInput"
  >
    labeled
  </label>
  <div>
    <div
      className="uu-tagInput"
      onClick={[Function]}
      role="none"
    >
      <span
        className="uu-multiSelectValue"
      >
        <div
          className="uu-multiSelectValueText"
        >
          <span
            className="uu-tagInputTagValue"
          >
            foo
          </span>
        </div>
        <button
          className="uu-multiSelectValueClose"
          onClick={[Function]}
          type="button"
        >
          <svg>
            icon-close.svg
          </svg>
        </button>
      </span>
      <span
        className="uu-multiSelectValue"
      >
        <div
          className="uu-multiSelectValueText"
        >
          <span
            className="uu-tagInputTagValue"
          >
            bar
          </span>
        </div>
        <button
          className="uu-multiSelectValueClose"
          onClick={[Function]}
          type="button"
        >
          <svg>
            icon-close.svg
          </svg>
        </button>
      </span>
      <span
        className="uu-multiSelectValue"
      >
        <div
          className="uu-multiSelectValueText"
        >
          <span
            className="uu-tagInputTagValue"
          >
            baz
          </span>
        </div>
        <button
          className="uu-multiSelectValueClose"
          onClick={[Function]}
          type="button"
        >
          <svg>
            icon-close.svg
          </svg>
        </button>
      </span>
      <input
        className="uu-tagInputNewItem"
        id="tagInput"
        name="testName"
        onBlur={[Function]}
        onKeyDown={[Function]}
        onPaste={[Function]}
      />
    </div>
    <button
      className="uu-button uu-button_view_dangerLink"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg>
        reset-icon.svg
      </svg>
      <span>
        Сбросить значения
      </span>
    </button>
  </div>
</div>
`;

exports[`UIKIT | TagInput Snapshots should labeled row 1`] = `
<div
  className="uu-labeled_direction_column"
>
  <label
    aria-label="labeled"
    className="uu-labeledLabel_direction_column uu-label uu-label_text"
    htmlFor="tagInput"
  >
    labeled
  </label>
  <div>
    <div
      className="uu-tagInput"
      onClick={[Function]}
      role="none"
    >
      <span
        className="uu-multiSelectValue"
      >
        <div
          className="uu-multiSelectValueText"
        >
          <span
            className="uu-tagInputTagValue"
          >
            foo
          </span>
        </div>
        <button
          className="uu-multiSelectValueClose"
          onClick={[Function]}
          type="button"
        >
          <svg>
            icon-close.svg
          </svg>
        </button>
      </span>
      <span
        className="uu-multiSelectValue"
      >
        <div
          className="uu-multiSelectValueText"
        >
          <span
            className="uu-tagInputTagValue"
          >
            bar
          </span>
        </div>
        <button
          className="uu-multiSelectValueClose"
          onClick={[Function]}
          type="button"
        >
          <svg>
            icon-close.svg
          </svg>
        </button>
      </span>
      <span
        className="uu-multiSelectValue"
      >
        <div
          className="uu-multiSelectValueText"
        >
          <span
            className="uu-tagInputTagValue"
          >
            baz
          </span>
        </div>
        <button
          className="uu-multiSelectValueClose"
          onClick={[Function]}
          type="button"
        >
          <svg>
            icon-close.svg
          </svg>
        </button>
      </span>
      <input
        className="uu-tagInputNewItem"
        id="tagInput"
        name="testName"
        onBlur={[Function]}
        onKeyDown={[Function]}
        onPaste={[Function]}
      />
    </div>
    <button
      className="uu-button uu-button_view_dangerLink"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg>
        reset-icon.svg
      </svg>
      <span>
        Сбросить значения
      </span>
    </button>
  </div>
</div>
`;

exports[`UIKIT | TagInput Snapshots should with error 1`] = `
<div>
  <div
    className="uu-tagInput uu-input_hasError"
    onClick={[Function]}
    role="none"
  >
    <span
      className="uu-multiSelectValue"
    >
      <div
        className="uu-multiSelectValueText"
      >
        <span
          className="uu-tagInputTagValue"
        >
          foo
        </span>
      </div>
      <button
        className="uu-multiSelectValueClose"
        onClick={[Function]}
        type="button"
      >
        <svg>
          icon-close.svg
        </svg>
      </button>
    </span>
    <span
      className="uu-multiSelectValue"
    >
      <div
        className="uu-multiSelectValueText"
      >
        <span
          className="uu-tagInputTagValue"
        >
          bar
        </span>
      </div>
      <button
        className="uu-multiSelectValueClose"
        onClick={[Function]}
        type="button"
      >
        <svg>
          icon-close.svg
        </svg>
      </button>
    </span>
    <span
      className="uu-multiSelectValue"
    >
      <div
        className="uu-multiSelectValueText"
      >
        <span
          className="uu-tagInputTagValue"
        >
          baz
        </span>
      </div>
      <button
        className="uu-multiSelectValueClose"
        onClick={[Function]}
        type="button"
      >
        <svg>
          icon-close.svg
        </svg>
      </button>
    </span>
    <input
      className="uu-tagInputNewItem"
      id="tagInput"
      name="testName"
      onBlur={[Function]}
      onKeyDown={[Function]}
      onPaste={[Function]}
    />
    <div
      className="uu-inputError"
    >
      <svg
        className="uu-inputErrorIcon uu-tooltipRoot uu-popoverRoot"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        error.svg
      </svg>
    </div>
  </div>
  <button
    className="uu-button uu-button_view_dangerLink"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <svg>
      reset-icon.svg
    </svg>
    <span>
      Сбросить значения
    </span>
  </button>
</div>
`;
